src/Infrastructure/Aggregate/PreconditionsTree/BeginnerPreconditions.php

```diff
@@ -4,14 +4,14 @@
 
 use PhpTuf\ComposerStager\Domain\Aggregate\PreconditionsTree\BeginnerPreconditionsInterface;
 use PhpTuf\ComposerStager\Domain\Aggregate\PreconditionsTree\CommonPreconditionsInterface;
-use PhpTuf\ComposerStager\Domain\Service\Precondition\CodebaseContainsNoSymlinksInterface;
+use PhpTuf\ComposerStager\Domain\Aggregate\PreconditionsTree\NoUnsupportedLinksExistInterface;
 use PhpTuf\ComposerStager\Domain\Service\Precondition\StagingDirDoesNotExistInterface;
 
 final class BeginnerPreconditions extends AbstractPreconditionsTree implements BeginnerPreconditionsInterface
 {
     public function __construct(
-        CodebaseContainsNoSymlinksInterface $codebaseContainsNoSymlinks,
         CommonPreconditionsInterface $commonPreconditions,
+        NoUnsupportedLinksExistInterface $noUnsupportedLinksExist,
```

Comment:
Even if it's not specified … Hyrum's Law says it will be relied upon anyway 😅 https://www.hyrumslaw.com/

I'd prefer this to be documented explicitly — any reason not to do so?

Reply:
Sure, we can make Hyrum happy. 😛 Added in the `Infrastructure\Aggregate\PreconditionsTree\AbstractPreconditionsTree::__construct()`:

> The order in which children are evaluated is unspecified and should not be depended upon. There is no guarantee that the order they are supplied in will have, or continue to have, any determinate effect.